### PR TITLE
Update and remove dependencies

### PR DESCRIPTION
**Pull request by ammar92**
### Changes
This PR updates and removes some unused and non-top-level dependencies.

Closes #2520 
Closes #2521 
Closes #2522 

Removes:
- Bytes
  - pathspec
  - platformdirs
  - astroid
  - importlib-resources
  - toml
  - mypy-extensions
  - mccabe
  - isort
  - attrs
- Rocky
  - ares
  - chardet
  - cwe
  - image
  - pyotp
  - toolz
  - wheel
- Octopoes
  - robotframework-tidy
---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: boefjes/pyproject.toml [hunk 1]
```diff
@@ -63,6 +62,7 @@ netaddr = "0.9.0"
 defusedxml = "^0.7.1"
 pydantic-settings = "^2.0.3"
 opentelemetry-instrumentation = "^0.41b0"
+pillow = "^10.2.0"
```
Comment (dekkers) [inline boefjes/pyproject.toml:65]: I think this is still only needed by the kat_webpage_analysis/check_images normalizer and should still be listed under that comment?

File: bytes/pyproject.toml [hunk 2]
```diff
@@ -57,23 +55,16 @@ opentelemetry-instrumentation-requests = "^0.41b0"
 opentelemetry-instrumentation = "^0.41b0"
 opentelemetry-instrumentation-dbapi = "^0.41b0"
 pydantic-settings = "^2.0.3"
+python-multipart = "^0.0.9"
```
Comment (dekkers) [inline bytes/pyproject.toml:58]: This shouldn't be a top-level dependency as far as I can see.

File: rocky/pyproject.toml [hunk 3]
```diff
@@ -62,17 +57,15 @@ opentelemetry-instrumentation = "^0.41b0"
 opentelemetry-instrumentation-fastapi = "^0.41b0"
 granian = "^1.0.2"
 pydantic-settings = "^2.0.3"
+pyparsing = "^3.1.1"
```
Comment (dekkers) [inline rocky/pyproject.toml:60]: I also can't find a reference to this in rocky, so I think this should also not be top-level dependency?

File: bytes/pyproject.toml (hunk 2 above)
Comment (ammar92) [reply bytes/pyproject.toml:58]: That's what I thought and removed it at first, but apparently FastAPI needed it so I reverted that action

File: rocky/pyproject.toml (hunk 3 above)
Comment (ammar92) [reply rocky/pyproject.toml:60]: The problem is the way how to "include" Octopoes. Pyparsing is used by Octopoes' path module, which is used by Rocky for the reports. This is something I wanted to discuss with you soon.

File: boefjes/pyproject.toml (hunk 1 above)
Comment (ammar92) [reply boefjes/pyproject.toml:65]: You're right, I accidentally moved that comment line. It should be good now

File: rocky/pyproject.toml (hunk 3 above)
Comment (dekkers) [reply rocky/pyproject.toml:60]: Ah you are right, I forgot about that...

File: bytes/pyproject.toml (hunk 2 above)
Comment (dekkers) [reply bytes/pyproject.toml:58]: I saw before that it was a dependency of FastAPI... but it is an optional dependency, so it it not installed by defautl.

Comment (ammar92) [reply bytes/pyproject.toml:58]: I know, but we use it. https://github.com/minvws/nl-kat-coordination/actions/runs/7922423908/job/21629977395

And I'm not sure if there's an alternative installation of FastAPI (e.g. `pip install fastapi[with-multipart]`)

**Review by underdarknl — APPROVED** [review]

**Pull request merged**